\documentclass[]{article}
\usepackage{lmodern}
\usepackage{amssymb,amsmath}
\usepackage{ifxetex,ifluatex}
\usepackage{fixltx2e} % provides \textsubscript
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
\else % if luatex or xelatex
  \ifxetex
    \usepackage{mathspec}
  \else
    \usepackage{fontspec}
  \fi
  \defaultfontfeatures{Ligatures=TeX,Scale=MatchLowercase}
\fi
% use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
% use microtype if available
\IfFileExists{microtype.sty}{%
\usepackage{microtype}
\UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\usepackage[margin=1in]{geometry}
\usepackage{hyperref}
\hypersetup{unicode=true,
            pdfborder={0 0 0},
            breaklinks=true}
\urlstyle{same}  % don't use monospace font for urls
\usepackage{longtable,booktabs}
\usepackage{graphicx,grffile}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
\IfFileExists{parskip.sty}{%
\usepackage{parskip}
}{% else
\setlength{\parindent}{0pt}
\setlength{\parskip}{6pt plus 2pt minus 1pt}
}
\setlength{\emergencystretch}{3em}  % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{0}
% Redefines (sub)paragraphs to behave more like sections
\ifx\paragraph\undefined\else
\let\oldparagraph\paragraph
\renewcommand{\paragraph}[1]{\oldparagraph{#1}\mbox{}}
\fi
\ifx\subparagraph\undefined\else
\let\oldsubparagraph\subparagraph
\renewcommand{\subparagraph}[1]{\oldsubparagraph{#1}\mbox{}}
\fi

%%% Use protect on footnotes to avoid problems with footnotes in titles
\let\rmarkdownfootnote\footnote%
\def\footnote{\protect\rmarkdownfootnote}

%%% Change title format to be more compact
\usepackage{titling}

% Create subtitle command for use in maketitle
\providecommand{\subtitle}[1]{
  \posttitle{
    \begin{center}\large#1\end{center}
    }
}

\setlength{\droptitle}{-2em}

  \title{}
    \pretitle{\vspace{\droptitle}}
  \posttitle{}
    \author{}
    \preauthor{}\postauthor{}
    \date{}
    \predate{}\postdate{}
  
\usepackage{amsmath}
\usepackage{graphicx}
\usepackage{booktabs}
\usepackage{lineno}
\usepackage{url}
% \usepackage[round]{natbib}
\bibliographystyle{apacite}
\usepackage[natbibapa]{apacite} 
\usepackage{fullpage} % set margins to be 1 inch around the page
\graphicspath{{../results/}}
\usepackage{booktabs}
\usepackage{longtable}
\usepackage{array}
\usepackage{multirow}
\usepackage{wrapfig}
\usepackage{float}
\usepackage{colortbl}
\usepackage{pdflscape}
\usepackage{tabu}
\usepackage{threeparttable}
\usepackage{threeparttablex}
\usepackage[normalem]{ulem}
\usepackage{makecell}
\usepackage{xcolor}

\begin{document}

\subsection{Supplemental material for:}

\section{\texorpdfstring{Movement of a \textit{Heliconius} hybrid zone
over 30 years: a Bayesian
approach}{Movement of a  hybrid zone over 30 years: a Bayesian approach}}\label{movement-of-a-hybrid-zone-over-30-years-a-bayesian-approach}

\subsection{Supplemental Methods}\label{supplemental-methods}

\subsubsection{Details on collection
data}\label{details-on-collection-data}

For Mallet's collections, we used the locality and phenotype data
presented in Table 4 of \citet{Mallet:1986vj}. Mallet sampled 20 sites
across Panama, but some of these sites were not on the main transect he
used to estimate cline parameters. We included 15 of Mallet's sites in
our analysis. We excluded 5 sites (which Mallet also excluded) for being
on the coast and too far from the main transect, on islands off the
mainland, or too far east. However, we included two sites (Río Iglesias
and Madden Dam) that Mallet excluded from his cline estimates.

For Blum's collections, we extracted locality and phenotype data from
Table 1 of \citet{Blum:2002wr}. Blum collected at 24 sites, but did not
include all sites in his calculation of the cline. We included 22 sites
in our analysis, excluding 2 sites (which Blum also excluded) for being
on islands off the mainland. We included two sites (Pipeline Road and
Madden Dam) that Blum excluded from his cline estimates. Blum did not
use all four phenotypic categories, only \textit{H. e. hydara},
\textit{H. e. demophoon}, and heterozygotes.

\textbf{N.B.}- the taxonomy of \textit{H. erato} has changed since the
publication of these earlier papers. \textit{H. erato demophoon} in
Panamá was previously considered \textit{H. erato petiverana}, and goes
by that older name in \citet{Mallet:1986vj} and \citet{Blum:2002wr}.

For the 2015 collections, at some sites we collected at multiple
subsites to achieve sufficient sample sizes. Before fitting clines, we
tested for genetic differentiation between subsites at the same site
using Fisher's exact tests on allele counts (pooling the
\textit{Cr\textsubscript{WC}} and \textit{Cr\textsubscript{CA}}
alleles). We found no evidence of differentiation (all \(P\)
\textgreater{} 0.05, results not shown), and thus combined subsites,
using the GPS coordinates of the subsite with the most samples or
randomly selected a subsite if sampling was equal.

\subsubsection{Phenotyping, genotyping, and estimation of allele
frequencies}\label{phenotyping-genotyping-and-estimation-of-allele-frequencies}

In Panamá there are three alleles at the \textit{Cr} locus, which we
designate: (1) \textit{Cr\textsubscript{HYD}}, the dominant,
black-hindwing allele found in \textit{H. e. hydara}; (2)
\textit{Cr\textsubscript{WC}}, the recessive, ventral-only yellow allele
found in the west Colombian \textit{H. e. venus}; and (3)
\textit{Cr\textsubscript{CA}}, the recessive, yellow allele found in the
Central American \textit{H. e. demophoon}. The dominance relationship
is: \textit{Cr\textsubscript{HYD}} is dominant to
\textit{Cr\textsubscript{WC}} which is dominant to
\textit{Cr\textsubscript{CA}} \citep{Mallet:1986vj}. Given this
dominance relationship, we can assign genotypes to the four phenotypic
classifications:

\begin{itemize}

\item[] (A) the north Colombian race, \textit{H. e. hydara}, with fully black hindwings, is homozygous for \textit{Cr\textsubscript{HYD}}.

\item[] (B) Heterozygotes, with black hindwings that display a faint yellow bar on the ventral side, could be either \textit{Cr\textsubscript{HYD}}/\textit{Cr\textsubscript{CA}} or \textit{Cr\textsubscript{HYD}}/\textit{Cr\textsubscript{WC}}. These genotypes cannot be distinguished visually. 


\item[] (C) the west Colombian race, \textit{H. e. venus}, with the yellow hindwing band present only on the ventral side, could be either \textit{Cr\textsubscript{WC}}/\textit{Cr\textsubscript{CA}} or \textit{Cr\textsubscript{WC}}/\textit{Cr\textsubscript{WC}}. These genotypes cannot be distinguished visually.


\item[] (D) the Central American race, \textit{H. e. demophoon}, with the yellow hindwing band on the dorsal and ventral sides, is homozygous for \textit{Cr\textsubscript{CA}}.

\end{itemize}

Given these phenotype possibilities, the frequency of the
\textit{Cr\textsubscript{HYD}} allele can be directly observed as:

\begin{equation}
f(Cr_{HYD}) = \frac{2A + B}{A + B + C +D}
\end{equation}

The combined frequency of the yellow alleles, \(f(Cr_{yel})\), is simply
\(1 - f(Cr_{hyd})\). However, determining which yellow allele(s) are
present in a population and calculating their frequencies is less
straightforward. For populations where \(f(Cr_{yel}) > 0\), there are 4
possible situations. When neither yellow homozygote is present in the
populations, we assume that any heterozygotes have the \(Cr_{CA}\)
allele, as this is more common in our study populations.

In populations with both types of yellow allele (e.g., with both type C
and type D individuals), the frequencies of the individual yellow
alleles cannot be directly observed. The appendix of Mallet (1986)
presents a maximum likelihood method for partitioning \(f(Cr_{yel})\)
into \(f(Cr_{CA})\) and \(f(Cr_{WC})\). Assuming the locus is at
Hardy-Weinberg equilibrium, the ratio between \(f(Cr_{CA})\) and
\(f(Cr_{WC})\) can be calculated as:

\begin{equation}
\frac{f(Cr_{CA})}{f(Cr_{WC})}  =  \frac{f_{D}+f_{D}(f_{C}+f_{D})}{f_{C}}
\end{equation}

where \(f(x)\) is the frequency of allele \(x\) and \(f_{X}\) is the
frequency of phenotype \(X\). Allele frequencies must add up to one such
that

\begin{equation}
f(Cr_{HYD}) + f(Cr_{WC}) + f(Cr_{CA}) = 1
\end{equation}

We can solve equation 2 for \(f(Cr_{WC})\):

\begin{equation}
f(Cr_{WC}) = \frac{f(Cr_{CA})f_{C}}{f_{D}+f_{D}(f_{C}+f_{D})}
\end{equation}

We can then substitue this result into equation 3 and do some algebra to
derive an equation for the frequency of the central American allele:

\begin{equation}
f(Cr_{CA}) = \frac{[f_{D}+f_{D}(f_{C}+f_{D})](1-f(Cr_{HYD}))}{f_{C} + f_{D}+f_{D}f_{C}+f_{D}^{2}}
\end{equation}

Using this equation, we can calculate allele frequencies for all three
alleles. Blum (2002) did not distinguish between West Columbian alleles
and central American alleles. But, applying this method to the
collections data from Mallet (1986) and this paper, we find that the
west Colombian \textit{Cr\textsubscript{WC}} allele is rare in our study
(tables S3 and S4). Thus, for this study we focus on the dominant
\textit{Cr\textsubscript{HYD}} allele, which can be directly observed,
and pool the rare \textit{Cr\textsubscript{WC}} yellow allele with the
more common \textit{Cr\textsubscript{CA}} yellow allele, as they cannot
be visually distinguished in heterozygotes.

\subsubsection{Assembling transect}\label{assembling-transect}

To calculate the one-dimensional distance along the transect, we
calculated the arclength of the cubic transect accounting for both the
curvature of the transect and the curvature of the earth. To calculate
the distance between \(a\) and \(b\), we evaluate:

\begin{equation}
\label{eq:dist}
R\frac{\pi}{180}\int_{a}^{b}\sqrt{\left(\cos\frac{f(x)\pi}{180}\right)^2+
f'(x)^2}\,dx
\end{equation}

where \(R\) is the radius of the earth at the equator, in km, \(a\) and
\(b\) are longitudes for the two sites, and \(f(x)\) is the equation
describing the cubic transect. For our calculations, we used
\(R = 6378.137\).

\subsubsection{Cline model equations}\label{cline-model-equations}

There are a number of parameterizations for cline introgression tails,
with slightly differing equations. We use the parameterization from
\citet{Gay:2008jp}, though our equations are modified to work with
clines of increasing allele frequency. We use three equations: one to
describe the left tail, one to describe the cline center, and one to
describe the right tail. For each model, we include introgression tails
as necessary, otherwise the equation for cline center is used to
describe the cline. For the mirrored tail model the parameters for the
left and right tails are equal, such that
\(\delta_{L} = \delta_{R} = \delta_{M}\) and
\(\tau_{L} = \tau_{R} = \tau_{M}\). The equations are:

Equation for left tail, (when \(x_{i} \leq c - \delta_{L}\)):

\begin{equation}
\label{eq:lefttail}
p_{i} =  p_{min} + (p_{max} - p_{min})\frac{1}{1 + e^{{4\frac{\delta_{L}}{w}}}}\exp{\frac{4\tau_{L}(x_{i}-c+\delta_{L})/w}{1 + e^{\frac{-4\delta_{L}}{w}}}}
\end{equation}

Equation for center (when \(c - \delta_{L} < x_{i} < c + \delta_{R}\)):

\begin{equation}
\label{eq:clineCenter}
p_{i} = p_{min} + (p_{max} - p_{min})\frac{e^{4\frac{(x_{i}-c)}{w}}}{1 + e^{4\frac{(x_{i}-c)}{w}}}
\end{equation}

Equation for right tail (when \(x_{i} \geq c + \delta_{R}\)):

\begin{equation}
\label{eq:righttail}
p_{i} =  p_{min} + (p_{max} - p_{min})\left(1-\frac{1}{1 + e^{{4\frac{\delta_{R}}{w}}}}\exp{\frac{-4\tau_{R}(x_{i}-c-\delta_{R})/w}{1 + e^{\frac{-4\delta_{R}}{w}}}}\right) 
\end{equation}

\subsubsection{Simulated data and model
validation}\label{simulated-data-and-model-validation}

To test our model, we simulated genotypic data from clines and compared
our model estimates to the simulated parameters. For each simulated
collection site, we used the cline equation without introgression tails
(equation 3 in the main text, equation 8 above) to calculate the
expected allele frequency, \(p\), at that site. Then, following equation
2 from the main text, we calculated predicted genotype frequencies given
the allele frequency, \(p\), and the simulated level of inbreeding,
\(F_{IS}\). From these genotype frequencies, we simulated genotypes of
diploid individuals by drawing from the multinomial distribution of
genotype frequencies, following equation 1 from the main text. Each
simulated dataset consisted of 41 collection sites spread at 10km
intervals from 0 to 400km along a transect, with 40 individuals
collected at each site.

We simulated datasets under a variety of parameters. We held the center
of the cline constant at 200km while varying the other parameters: cline
width of 20km and 80km, \(p_{min}\) of 0.04 and 0.15, \(p_{max}\) of
0.85 and 0.97, and \(F_{IS}\) of 0, 0.1, 0.25, 0.5, 0.75, and 1. There
were thus 48 different possible parameter combinations. For each
parameter combination we simulated 15 datasets, for a total of 720
simulated datasets.

For each simulated dataset, we fit the cline models two ways: (1) using
our novel Bayesian model, (2) using a maximum likelihood approach in the
\texttt{R} package \texttt{HZAR} \citep{Derryberry:2014jw} and applying
the effective sample size correction of \citet{Alexandrino:2005vl}. We
refer to these approaches as (1) Bayesian and (2) corrected ML.

For the Bayesian approach, we fit the cline model without introgression
tails in \texttt{Stan} v2.17.0 and \texttt{RStan} v2.17.3
\citep{Carpenter:2017ke, Anonymous:tt}. We placed weak normal priors on
the center \(N(350, 100)\) and width \(N(50, 100)\), both constrained to
be positive. For \(p_{min}\) and \(p_{max}\), we used uniform priors of
\(U(0,0.2)\) and \(U(0.8,1)\), respectively. We fit four independent
chains with 3000 iterations of warm-up and 7000 iterations of sampling,
for a total of 28000 samples from the posterior distribution. Chains
were run in parallel across 4 processor cores. We generated point
estimates and credible intervals for each parameter using the mean and
95\% highest posterior density interval (HPDI) of the marginal posterior
distribution of each parameter.

For the corrected ML approach, we fit models in \texttt{HZAR} following
the example code given in appendix 1 of \citealt{Derryberry:2014jw}, but
removing unnecessary visualization steps to speed model fitting. We fit
only the model without introgression tails (``free.none''). We used
default settings for all functions and ran chains in parallel. However,
we modified the initialization values for all parameters. The models
would often fail to fit using the default initialization values, so
instead we drew random starting values for each parameter from the same
distributions we used to initialize our Bayesian models: for center, a
normal distribution with mean equal to the simulated value and a
standard deviation of 20; for width, a normal distribution with mean
equal to the simulated value and a standard deviation of 15; and for
\(p_{min}\) and \(p_{max}\) uniform distributions of \(U(0,0.2)\) and
\(U(0.8,1)\), respectively. For each parameter, we used the ML value as
the point estimate and the lower and upper two-unit log-likelihood
limits as the lower and upper confidence intervals
\citep{Derryberry:2014jw}.

We used the \texttt{R} package \texttt{tictoc} to time each individual
instance of model fitting and compare across methods \citep{tictoc:tt}.
All model fitting was done on a Mac Pro with a 3-Ghz, 8 core Intel Xeon
processor with 64 GB of RAM. The average time to fit a model using our
Bayesian approach was 30 seconds, while the average runtime for the
corrected ML approach was 5.81 minutes.

To compare model accuracy, we calculated the root-mean-square deviation
(RMSD) between the estimated parameter values from our models and the
simulated values for each of the 48 combinations of parameters. RMSD is
a measure of accuracy, with lower values indicating a more accurate
model (i.e., smaller average squared differences between the estimated
parameter value and the simulated parameter value). For each of the 48
parameter combinations, we calculated the RMSD of each model and for
each cline parameter (RMSD is scale-dependent and cannot be compared
across parameters).

We used paired t-tests to examine whether the average RMSD across
parameter combinations differed between our Bayesian approach and the
corrected ML approach. Our Bayesian approach had a significantly smaller
average RMSD (was more accurate) than the corrected ML approach for the
center and width parameters (Table S2). However, these differences in
accuracy were relatively small. For the \(p_{min}\) and \(p_{max}\)
parameters the difference in average RMSD between the models was not
statistically different from 0 (Table S2).

As another measure of model accuracy, we also calculated how often the
``true'' simulated value of a parameter was included within the
confidence intervals estimated by the model. For our Bayesian method,
the true simulated value fell within the 95\% credible intervals 93.33\%
of the time. This is a slight improvement over the ML model (92.71\%),
and indicates that our method better models uncertainty around parameter
estimates.

\subsubsection{Forest data}\label{forest-data}

We extracted forest data for Panamá from v1.5 of the Global Forest
Change dataset of \citealt{Hansen:2013iy}, found at
\url{https://earthenginepartners.appspot.com/science-2013-global-forest/download_v1.5.html}.
We downloaded six files:

\begin{longtable}[]{@{}lll@{}}
\toprule
Filename & Year & Data\tabularnewline
\midrule
\endhead
Hansen\_GFC-2017-v1.5\_first\_10N\_080W.tif & 2000 & Landsat
multispectral\tabularnewline
Hansen\_GFC-2017-v1.5\_first\_10N\_090W.tif & 2000 & Landsat
multispectral\tabularnewline
Hansen\_GFC-2017-v1.5\_last\_10N\_080W.tif & 2017 & Landsat
multispectral\tabularnewline
Hansen\_GFC-2017-v1.5\_last\_10N\_090W.tif & 2017 & Landsat
multispectral\tabularnewline
Hansen\_GFC-2017-v1.5\_lossyear\_10N\_080W.tif & 2000-2017 & forest
loss\tabularnewline
Hansen\_GFC-2017-v1.5\_lossyear\_10N\_090W.tif & 2000-2017 & forest
loss\tabularnewline
\bottomrule
\end{longtable}

For each dataset, we used \texttt{QGIS} \citep{Anonymous:qg} to merge
the files for east Panamá (80W) and west Panamá (90W) together, and to
crop the images down to include only Panamá.

The \citealt{Hansen:2013iy} forest loss images encode the data for each
pixel as either 0 (no forest loss, where loss is ``defined as a
stand-replacement disturbance'') or a number from 1-17, representing the
year of major forest loss. Thus, to determine the proportion of forest
lost within a given area, we calculated 1-(number of pixels with value
of 0/total number of pixels). We made those calculations using the
\texttt{raster} package in \texttt{R} \citep{raster:rp}.

The \citealt{Hansen:2013iy} Landsat multispectral images contain data
from four bands, with 8-bit, normalized top-of-atmosphere reflectance
values for each band (\(\rho\)). For NDVI calculation, we used band 3
(red) and band 4 (near infrared, NIR) to calculate NDVI as:

\begin{equation}
\label{eq:ndvi}
NDVI = \frac{\rho_{NIR}-\rho_{red}}{\rho_{NIR}+\rho_{red}}
\end{equation}

We calculated NDVI separately for each year (2000 and 2017), using the
raster calculator in \texttt{QGIS} \citep{Anonymous:qg} to calculate
NDVI and \(\Delta\)NDVI from 2000 to 2017 (i.e., \(\Delta\)NDVI =
NDVI\(_{2017}\) - NDVI\(_{2000}\)). To find the mean NDVI or mean
\(\Delta\)NDVI within a given area, we used the \texttt{raster} package
in \texttt{R} \citep{raster:rp}.

\bibliography{hz}

\pagebreak

\subsection{Supplemental tables and
figures}\label{supplemental-tables-and-figures}

\subsubsection{Table S1- Butterflies collected in
2015}\label{table-s1--butterflies-collected-in-2015}

\begin{table}[H]
\centering
\begin{tabular}{lrrrrrrr}
\toprule
site.collected & coord.N.decdeg & coord.W.decdeg & A.melanized & B.hetero & C.west.col & D.postman & total\\
\midrule
el valle & 8.593233 & -80.14108 & 3 & 0 & 0 & 19 & 22\\
cerro campana & 8.704150 & -79.89185 & 0 & 1 & 0 & 34 & 35\\
sherman & 9.248100 & -79.94779 & 0 & 3 & 0 & 28 & 31\\
gamboa & 9.116050 & -79.69837 & 0 & 0 & 0 & 35 & 35\\
tocumen & 9.201050 & -79.39247 & 0 & 4 & 0 & 33 & 37\\
\addlinespace
tapagra & 9.166333 & -79.20790 & 3 & 5 & 0 & 9 & 17\\
el llano & 9.237533 & -78.95347 & 8 & 16 & 1 & 9 & 34\\
loma naranjo & 9.179817 & -78.88455 & 3 & 10 & 0 & 6 & 19\\
corp. bayano & 9.207400 & -78.82647 & 1 & 8 & 0 & 14 & 23\\
mangowichi & 9.137667 & -78.68952 & 20 & 12 & 1 & 3 & 36\\
\addlinespace
ipeti & 8.972917 & -78.51060 & 15 & 6 & 0 & 0 & 21\\
agua fria & 8.858850 & -78.22670 & 15 & 8 & 1 & 1 & 25\\
casa pastoral & 8.663683 & -78.16152 & 17 & 4 & 1 & 1 & 23\\
puertolara & 8.613550 & -78.13982 & 45 & 5 & 0 & 0 & 50\\
meteti & 8.406767 & -77.99898 & 32 & 6 & 0 & 0 & 38\\
\addlinespace
IFAD & 8.304800 & -77.81590 & 28 & 2 & 0 & 0 & 30\\
yaviza & 8.204617 & -77.71825 & 27 & 6 & 0 & 0 & 33\\
\bottomrule
\end{tabular}
\end{table}

\textbf{Table S1}- Site names, GPS coordinates (in decimal degrees), and
number of samples from each phenotypic class (as defined by
\citealt{Mallet:1986vj}, see supplemental methods).

\pagebreak

\subsubsection{Table S2- Comparison of model
accuracy}\label{table-s2--comparison-of-model-accuracy}

\begin{table}[H]
\centering
\begin{tabular}{lrrrr}
\toprule
parameter & difference in RMSD & T statistic & Degrees of freedom & P\\
\midrule
center & -0.022 & -2.435 & 47 & 0.019\\
width & -0.156 & -2.037 & 47 & 0.047\\
pmin & 0.000 & 1.416 & 47 & 0.163\\
pmax & 0.000 & 0.424 & 47 & 0.674\\
\bottomrule
\end{tabular}
\end{table}

\textbf{Tables S2}- Results of paired \textit{t}-tests comparing the
RMSD for each model. When difference in RMSD is negative the Bayesian
model has a smaller RMSD (is more accurate).

\pagebreak 

\subsubsection{Table S3- Frequencies for all three alleles, Mallet
(1986)
Collections}\label{table-s3--frequencies-for-all-three-alleles-mallet-1986-collections}

\begin{table}[H]
\centering
\begin{tabular}{rlrrr}
\toprule
Year & Site & f.HYD & f.CA & f.WC\\
\midrule
1982 & El Copé & 0.07 & 0.93 & 0.00\\
1982 & Madden Dam & 0.00 & 1.00 & 0.00\\
1982 & Ciudad Panamá & 0.01 & 0.99 & 0.00\\
1982 & Tocumen & 0.15 & 0.85 & 0.00\\
1982 & El Llano-Cartí & 0.05 & 0.95 & 0.00\\
\addlinespace
1982 & Bayano & 0.04 & 0.96 & 0.00\\
1982 & Piriatí & 0.12 & 0.88 & 0.00\\
1982 & Near Tortí & 0.35 & 0.65 & 0.00\\
1982 & Cañazas & 0.67 & 0.31 & 0.02\\
1982 & Quebrada Mono & 0.89 & 0.11 & 0.00\\
\addlinespace
1982 & Río Iglesias & 0.85 & 0.15 & 0.00\\
1982 & Meteí & 0.91 & 0.09 & 0.00\\
1982 & Canglón & 0.95 & 0.05 & 0.00\\
1982 & Near Yaviza & 0.91 & 0.09 & 0.00\\
1982 & Cana & 0.95 & 0.05 & 0.00\\
\bottomrule
\end{tabular}
\end{table}

\subsubsection{Table S4- Frequencies for all three alleles, 2015
Collections}\label{table-s4--frequencies-for-all-three-alleles-2015-collections}

\begin{table}[H]
\centering
\begin{tabular}{rlrrr}
\toprule
Year & Site & f.HYD & f.CA & f.WC\\
\midrule
2015 & el valle & 0.14 & 0.86 & 0.00\\
2015 & cerro campana & 0.01 & 0.99 & 0.00\\
2015 & sherman & 0.05 & 0.95 & 0.00\\
2015 & gamboa & 0.00 & 1.00 & 0.00\\
2015 & tocumen & 0.05 & 0.95 & 0.00\\
\addlinespace
2015 & tapagra & 0.32 & 0.68 & 0.00\\
2015 & el llano & 0.47 & 0.49 & 0.04\\
2015 & loma naranjo & 0.42 & 0.58 & 0.00\\
2015 & corp. bayano & 0.22 & 0.78 & 0.00\\
2015 & mangowichi & 0.72 & 0.21 & 0.06\\
\addlinespace
2015 & ipeti & 0.86 & 0.14 & 0.00\\
2015 & agua fria & 0.76 & 0.12 & 0.12\\
2015 & casa pastoral & 0.83 & 0.09 & 0.08\\
2015 & puertolara & 0.95 & 0.05 & 0.00\\
2015 & meteti & 0.92 & 0.08 & 0.00\\
\addlinespace
2015 & IFAD & 0.97 & 0.03 & 0.00\\
2015 & yaviza & 0.91 & 0.09 & 0.00\\
\bottomrule
\end{tabular}
\end{table}

\textbf{Tables S3-S4}- Maximum-likelihood estimates of the allele
frequencies of the three \textit{Cr} alleles in the 1982 and 2015
samples.

\pagebreak

\subsubsection{Table S5- Parameter estimates for all tail models, 1982
cline}\label{table-s5--parameter-estimates-for-all-tail-models-1982-cline}

\begin{table}[H]
\centering\begingroup\fontsize{9}{11}\selectfont

\begin{tabular}{lrrrrr}
\toprule
\multicolumn{1}{c}{ } & \multicolumn{5}{c}{Introgression tails} \\
\cmidrule(l{3pt}r{3pt}){2-6}
parameter & none & left & right & mirror & ind\\
\midrule
center & 516.05 & 516.42 & 516.43 & 516.54 & 516.67\\
width & 52.87 & 49.42 & 50.96 & 48.64 & 46.99\\
pmin & 0.07 & 0.06 & 0.07 & 0.06 & 0.06\\
pmax & 0.91 & 0.91 & 0.93 & 0.92 & 0.93\\
deltaL & NA & 30.43 & NA & NA & 29.10\\
tauL & NA & 0.62 & NA & NA & 0.62\\
deltaR & NA & NA & 26.94 & NA & 25.22\\
tauR & NA & NA & 0.59 & NA & 0.59\\
deltaM & NA & NA & NA & 30.91 & NA\\
tauM & NA & NA & NA & 0.61 & NA\\
\bottomrule
\end{tabular}
\endgroup{}
\end{table}

\subsubsection{Table S6- Parameter estimates for all tail models, 1999
cline}\label{table-s6--parameter-estimates-for-all-tail-models-1999-cline}

\begin{table}[H]
\centering\begingroup\fontsize{9}{11}\selectfont

\begin{tabular}{lrrrrr}
\toprule
\multicolumn{1}{c}{ } & \multicolumn{5}{c}{Introgression tails} \\
\cmidrule(l{3pt}r{3pt}){2-6}
parameter & none & left & right & mirror & ind\\
\midrule
center & 467.03 & 466.48 & 467.15 & 466.88 & 466.59\\
width & 63.59 & 60.67 & 59.52 & 59.13 & 56.45\\
pmin & 0.04 & 0.03 & 0.04 & 0.03 & 0.03\\
pmax & 0.92 & 0.92 & 0.95 & 0.94 & 0.94\\
deltaL & NA & 31.82 & NA & NA & 30.60\\
tauL & NA & 0.57 & NA & NA & 0.57\\
deltaR & NA & NA & 20.56 & NA & 19.66\\
tauR & NA & NA & 0.57 & NA & 0.56\\
deltaM & NA & NA & NA & 27.51 & NA\\
tauM & NA & NA & NA & 0.54 & NA\\
\bottomrule
\end{tabular}
\endgroup{}
\end{table}

\subsubsection{Table S7- Parameter estimates for all tail models, 2015
cline}\label{table-s7--parameter-estimates-for-all-tail-models-2015-cline}

\begin{table}[H]
\centering\begingroup\fontsize{9}{11}\selectfont

\begin{tabular}{lrrrrr}
\toprule
\multicolumn{1}{c}{ } & \multicolumn{5}{c}{Introgression tails} \\
\cmidrule(l{3pt}r{3pt}){2-6}
parameter & none & left & right & mirror & ind\\
\midrule
center & 450.09 & 450.00 & 450.89 & 451.06 & 451.37\\
width & 99.45 & 89.77 & 93.18 & 82.75 & 80.52\\
pmin & 0.04 & 0.03 & 0.04 & 0.03 & 0.03\\
pmax & 0.91 & 0.91 & 0.94 & 0.92 & 0.94\\
deltaL & NA & 44.23 & NA & NA & 35.57\\
tauL & NA & 0.69 & NA & NA & 0.70\\
deltaR & NA & NA & 24.81 & NA & 22.38\\
tauR & NA & NA & 0.66 & NA & 0.61\\
deltaM & NA & NA & NA & 32.23 & NA\\
tauM & NA & NA & NA & 0.71 & NA\\
\bottomrule
\end{tabular}
\endgroup{}
\end{table}

\textbf{Tables S5-S7}- Cline parameter estimates (posterior mean) for
all five possible tail models (no introgression tails, left tail, right
tail, mirrored tails, and independent tails) for each year.

\pagebreak

\subsubsection{Table S8- Model comparison, 1982
cline}\label{table-s8--model-comparison-1982-cline}

\begin{table}[H]
\centering
\begin{tabular}{lrrrr}
\toprule
  & WAIC & pWAIC & dWAIC & weight\\
\midrule
none & 102.80 & 8.57 & 0.00 & 0.28\\
right & 103.19 & 8.67 & 0.39 & 0.23\\
mirror & 103.58 & 8.77 & 0.78 & 0.19\\
left & 103.71 & 8.82 & 0.91 & 0.18\\
ind & 104.42 & 9.07 & 1.62 & 0.12\\
\bottomrule
\end{tabular}
\end{table}

\subsubsection{Table S9- Model comparison, 1999
cline}\label{table-s9--model-comparison-1999-cline}

\begin{table}[H]
\centering
\begin{tabular}{lrrrr}
\toprule
  & WAIC & pWAIC & dWAIC & weight\\
\midrule
right & 115.21 & 9.44 & 0.00 & 0.29\\
ind & 115.42 & 9.64 & 0.21 & 0.27\\
mirror & 115.56 & 9.66 & 0.36 & 0.25\\
left & 117.38 & 10.04 & 2.17 & 0.10\\
none & 117.50 & 9.95 & 2.29 & 0.09\\
\bottomrule
\end{tabular}
\end{table}

\subsubsection{Table S10- Model comparison, 2015
cline}\label{table-s10--model-comparison-2015-cline}

\begin{table}[H]
\centering
\begin{tabular}{lrrrr}
\toprule
  & WAIC & pWAIC & dWAIC & weight\\
\midrule
right & 157.64 & 15.37 & 0.00 & 0.48\\
none & 158.52 & 15.26 & 0.88 & 0.31\\
mirror & 161.29 & 16.82 & 3.65 & 0.08\\
left & 161.40 & 16.50 & 3.76 & 0.07\\
ind & 161.75 & 17.51 & 4.11 & 0.06\\
\bottomrule
\end{tabular}
\end{table}

\textbf{Tables S8-S10}- Table of WAIC comparisons and Akaike weights for
the five possible tail models (no introgression tails, left tail, right
tail, mirrored tails, and independent tails) for each year. pWAIC is the
effective number of parameters, dWAIC is the difference in WAIC compared
with the model with the lowest WAIC, and weight is the Akaike weight.

\pagebreak

\pagebreak

\subsubsection{Figure S1- MCMC diagnostics, 1982
Cline}\label{figure-s1--mcmc-diagnostics-1982-cline}

\begin{center}\includegraphics{supplemental_files/figure-latex/unnamed-chunk-2-1} \end{center}

\textbf{Figure S1}- Trace plots of the cline parameters for the best fit
model (no tails) for the 1982 cline

\pagebreak

\subsubsection{Figure S2- MCMC diagnostics, 1999
Cline}\label{figure-s2--mcmc-diagnostics-1999-cline}

\begin{center}\includegraphics{supplemental_files/figure-latex/unnamed-chunk-3-1} \end{center}

\textbf{Figure S2}- Trace plots of the cline parameters for the best fit
model (right tail) for the 1999 cline

\pagebreak 

\subsubsection{Figure S3- MCMC diagnostics, 2015
Cline}\label{figure-s3--mcmc-diagnostics-2015-cline}

\begin{center}\includegraphics{supplemental_files/figure-latex/unnamed-chunk-4-1} \end{center}

\textbf{Figure S3}- Trace plots of the cline parameters for the best fit
model (right tail) for the 2015 cline

\pagebreak

\subsubsection{Figure S4- Inbreeding
coefficients}\label{figure-s4--inbreeding-coefficients}

\begin{center}\includegraphics{supplemental_files/figure-latex/plot fis-1} \end{center}

\textbf{Figure S4}- Estimates of of the inbreeding coefficient,
\(F_{IS}\), across the hybrid zone. Points and lines show the mean
\(\pm\) 95\% HPDI of the posterior distribution of \(F_{IS}\) for each
site along the transect, as estimated from the best-fit cline model for
each year.

\pagebreak

\subsubsection{Figure S5- Posterior predictive check, 1982
cline}\label{figure-s5--posterior-predictive-check-1982-cline}

\begin{center}\includegraphics{supplemental_files/figure-latex/unnamed-chunk-5-1} \end{center}

\textbf{Figure S5}- Posterior predictive check of the best fit cline
model (no tails) for the 1982 cline. Within each panel, vertical lines
represent the 95\% posterior predictive interval of the expected
genotype frequency at that site, while points show the observed genotype
frequency. Genotypes are represented separately in each panel (top =
\(AA\), middle = \(Aa\), botom = \(aa\)). Sites colored in orange show
an observed genotype frequency outside of the 95\% posterior predictive
interval.

\pagebreak

\subsubsection{Figure S6- Posterior predictive check, 1999
cline}\label{figure-s6--posterior-predictive-check-1999-cline}

\begin{center}\includegraphics{supplemental_files/figure-latex/unnamed-chunk-6-1} \end{center}

\textbf{Figure S6}- Posterior predictive check of the best fit cline
model (right tail) for the 1999 cline. Within each panel, vertical lines
represent the 95\% posterior predictive interval of the expected
genotype frequency at that site, while points show the observed genotype
frequency. Genotypes are represented separately in each panel (top =
\(AA\), middle = \(Aa\), botom = \(aa\)). Sites colored in orange show
an observed genotype frequency outside of the 95\% posterior predictive
interval.

\pagebreak

\subsubsection{Figure S7- Posterior predictive check, 2015
cline}\label{figure-s7--posterior-predictive-check-2015-cline}

\begin{center}\includegraphics{supplemental_files/figure-latex/unnamed-chunk-7-1} \end{center}

\textbf{Figure S7}- Posterior predictive check of the best fit cline
model (right tail) for the 2015 cline. Within each panel, vertical lines
represent the 95\% posterior predictive interval of the expected
genotype frequency at that site, while points show the observed genotype
frequency. Genotypes are represented separately in each panel (top =
\(AA\), middle = \(Aa\), botom = \(aa\)). Sites colored in orange show
an observed genotype frequency outside of the 95\% posterior predictive
interval.

\pagebreak

\subsubsection{Figure S8- Forest cover across the
transect}\label{figure-s8--forest-cover-across-the-transect}

\begin{center}\includegraphics{supplemental_files/figure-latex/envFig-1} \end{center}

\textbf{Figure S8}- Variation in forest loss, NDVI, and change in NDVI
across Panamá. For all panels, each point represents one of 47 circles
of radius 5km at 15km intervals along our transect. Y-axes display
measurements of forest dynamics within each circle: (A) proportion of
forest loss, (B) mean NDVI in 2000, (C) mean NDVI in 2017, (D) mean
difference in NDVI from 2000 to 2017. Each panel includes a
loess-smoothed line to visualize trends. The vertical lines show the
estimated center of the hybrid zone in 2000 (dotted) and 2015 (dashed).


\end{document}
